packages/lesswrong/lib/collections/users/schema.ts

```diff
@@ -216,15 +216,15 @@ const schema: SchemaType<DbUser> = {
     onUpdate: async ({data, oldDocument}) => {
       if (data.slug && data.slug !== oldDocument.slug) {
         const slugLower = data.slug.toLowerCase();
-        const slugIsUsed = await Utils.slugIsUsed("Users", slugLower)
+        const slugIsUsed = await Utils.slugIsUsed("Users", slugLower, oldDocument._id)
```

Comment:
Instead of passing in the `_id` of the document to ignore, could we just check beforehand if the new slug is in `oldSlugs`, and if yes then we skip the call to `slugIsUsed()` and allow the new slug to be used? Or is it possible for some other user to have that slug in their `oldSlugs` as well?

Reply:
Yeah it is possible for another user to have the slug in their `oldSlugs` since `slugLower` is just an arbitrary string the user enters, so we still want to disallow the change in that case (to make sure old slugs to redirect to the correct user to preserve links). I think this is the only way to achieve this with one trip to the db

Reply:
If the old slug redirects to the user's current profile page, doesn't that mean that only one user could have the old slug in their `oldSlugs`? Basically it looks like the code tries to enforce that no two users have the same string in `oldSlugs`. Though maybe there are bugs and this has happened before?

Reply:
It is true that no two users can have the same slug in their old slugs, but that's not what this is trying to enforce. It's checking if `slugLower` (which can be anything the user typed in) already exists as:
 - any user's current slug (including the current user)
 - any user's old slug, apart from the current user, who we want to explicitly allow to switch back to one of their old slugs. That's what this line is for `{$and: [{ oldSlugs: slug }, { _id: { $ne: idToAllowOldSlugsFor }}]}`

Reply:
Sorry, I think I'm not explaining this clearly. By "the code tries to enforce...", I meant the codebase as a whole, not the code in this PR.

My overall suggestion is that we leave `slugIsUsed()` as it originally was. But before we check `slugIsUsed()`, we first just see if `slugLower` matches something in `oldSlugs` for this user, and if so we shortcut and allow them to use that slug.

So something like
```
if (data.slug && data.slug !== oldDocument.slug) {
  // pseudocode, prob we need to do a case-insensitive check
  if (slugLower in oldDocument.oldSlugs) return slugLower
  // the code below here is unchanged from master - at this point we know that slugLower is not the current user's slug, nor is it in their oldSlugs
  const slugIsUsed = await Utils.slugIsUsed("Users", slugLower)
  ...
}
```

Let me know if this is unclear.

Reply:
Ah ok, sorry I didn't understand your original comment, I thought you were referring to all `oldSlugs` not just `oldDocument.oldSlugs` 😌, I've changed it to what you suggested now